llms/maritaca/options.go

```diff
@@ -0,0 +1,166 @@
+package maritaca
+
+import (
+	"log"
+	"net/http"
+	"net/url"
+
+	"github.com/tmc/langchaingo/llms/maritaca/internal/maritacaclient"
+)
+
+type options struct {
+	maritacaServerURL   *url.URL
+	httpClient          *http.Client
+	model               string
+	maritacaOptions     maritacaclient.Options
+	customModelTemplate string
+	system              string
+	format              string
+}
+
+type Option func(*options)
+
+// WithModel Set the model to use.
+func WithModel(model string) Option {
+	return func(opts *options) {
+		opts.model = model
+	}
+}
+
+// WithFormat Sets the maritaca output format (currently maritaca only supports "json").
+func WithFormat(format string) Option {
+	return func(opts *options) {
+		opts.format = format
+	}
+}
+
+// WithSystem Set the system prompt. This is only valid if
+// WithCustomTemplate is not set and the maritaca model use
+// .System in its model template OR if WithCustomTemplate
+// is set using {{.System}}.
+func WithSystemPrompt(p string) Option {
+	return func(opts *options) {
+		opts.system = p
+	}
+}
+
+// WithCustomTemplate To override the templating done on maritaca model side.
+func WithCustomTemplate(template string) Option {
+	return func(opts *options) {
+		opts.customModelTemplate = template
+	}
+}
+
+// WithServerURL Set the URL of the maritaca instance to use.
+func WithServerURL(rawURL string) Option {
+	return func(opts *options) {
+		var err error
+		opts.maritacaServerURL, err = url.Parse(rawURL)
+		if err != nil {
+			log.Fatal(err)
+		}
+	}
+}
+
+// WithHTTPClient Set custom http client.
+func WithHTTPClient(client *http.Client) Option {
+	return func(opts *options) {
+		opts.httpClient = client
+	}
+}
+
+// default: true
```

Comment:
please start these with the name of the function being documented.

Reply:
did